### ISSUE TITLE
Drones and gravitygen-less tiles

### ISSUE THREAD

**Issue by Midaychi** [tgstation/-tg-station#10928]
Unless a drone happens to have a fire extinguisher with them, they're completely fucked if they get stranded on tiles on a level without a gravity gen.

They can't throw anything, anything they construct doesn't update their ability to move, they can't even trip balls over lockers. (I constructed a girder and a grille and a wall on the spot I was floating on and still couldn't move.)

How to test: Go to box wrecked telecomms as a drone, get stranded on a tile or in the middle of the abandoned teleporter without a fire extinguisher. Flail around helplessly no matter what you do for the rest of the round.

**Comment by Midaychi** [comment]
example: I couldn't move in this image. I built the wall after I got stuck.

![stuck](https://cloud.githubusercontent.com/assets/1968809/8968717/e2acbbf4-35f3-11e5-90db-e10b555fe6de.png)

Could fix by giving drones a built in emergency thruster, that can run out of fuel but be recharged somehow.  Activate thruster mode, then click a direction to be shot in the opposite direction and use whatever it uses as fuel.
